Modules/_zoneinfo.c

```diff
@@ -7,6 +7,17 @@
 
 #include "datetime.h"
 
+// This takes advantage of GCC compiler extensions to determine the type of the
+// variable, so we'll make Py_ASSERT_VAR_UNSIGNED a noop on non-GCC platforms.
+// This is not perfect, but it's better than nothing.
+#ifdef __GNUC__
```

Comment:
Hum, I would prefer to test a "HAVE_TYPEOF" rather than "hardcoding" GCC here. I suggest to define it in pymacros.h. Something like:

```
#if defined(__GNUC__) || defined(__clang__)
   // typeof() function
#  define HAVE_TYPEOF 1
#endif
```

clang also supports typeof().

Note: I prefer `typeof()` over `__typeof__()` :-)

typeof() is already used in pymacros.h. __typeof__() is used in pycore_atomic.h.

Reply:
Oh, clang manual says:

The parser recognizes “asm” and “typeof” as keywords in gnu* modes; the variants “__asm__” and “__typeof__” are recognized in all modes.

Reply:
Yeah, I think `__typeof__` is the "compatibility mode" version of this. The standard way you build extensions seems to use one of the modes where `typeof` doesn't work as far as I can tell, so that's how I did it there. If we're already unconditionally using `typeof()` I guess it's fine to switch to that here.